composition/src/normalization/walkers.ts

```diff
@@ -395,6 +394,9 @@ export function upsertParentsAndChildren(nf: NormalizationFactory, document: Doc
           nf.isSubgraphVersionTwo,
           nf.errors,
         );
+        if (isParentRootType) {
```

Comment:
Please could you add a comment that this line was moved so that we have access to argumentData?